include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm.hpp

```diff
@@ -0,0 +1,226 @@
+     * algorithm type.
+     *
+     * ### Complexity
+     *
+     * The following table lists the runtime and space complexities for the banded and unbanded algorithm dependent
+     * on the configured seqan3::align_cfg::result per sequence pair.
+     * Let `n` be the length of the first sequence, `m` be the length of the second sequence and `k` be the size of
+     * the band.
+     *
+     * |                        | unbanded         | banded            |
+     * |:----------------------:|:----------------:|:-----------------:|
+     * |runtime                 |\f$ O(n*m) \f$    |\f$ O(n*k) \f$     |
+     * |space (score only)      |\f$ O(m) \f$      |\f$ O(k) \f$       |
+     * |space (end positions)   |\f$ O(m) \f$      |\f$ O(k) \f$       |
+     * |space (begin positions) |\f$ O(n*m) \f$    |\f$ O(n*k) \f$     |
+     * |space (alignment)       |\f$ O(n*m) \f$    |\f$ O(n*k) \f$     |
+     */
+    template <indexed_sequence_pair_range indexed_sequence_pairs_t, typename callback_t>
+    //!\cond
+        requires std::invocable<callback_t, alignment_result_type>
+    //!\endcond
+    void operator()(indexed_sequence_pairs_t && indexed_sequence_pairs, callback_t && callback)
+    {
+        using result_value_t = typename alignment_result_value_type_accessor<alignment_result_type>::type;
+        using std::get;
+
+        for (auto && [sequence_pair, idx] : indexed_sequence_pairs)
+        {
+            result_value_t res{};
+            res.id = idx;
+            res.score = compute_matrix(get<0>(sequence_pair), get<1>(sequence_pair));
+            callback(alignment_result_type{res});
+        }
+    }
+
+protected:
+    /*!\brief Compute the actual alignment by given the two sequences.
+     * \tparam sequence1_t The type of the first sequence; must model std::ranges::forward_range.
+     * \tparam sequence2_t The type of the second sequence; std::ranges::forward_range.
+     *
+     * \param[in] sequence1 The first sequence.
+     * \param[in] sequence2 The second sequence.
+     */
+    template <std::ranges::forward_range sequence1_t, std::ranges::forward_range sequence2_t>
+    int32_t compute_matrix(sequence1_t & sequence1, sequence2_t & sequence2)
+    {
+        // Use gap costs locally.
+        int32_t gap_extension{m_gap_scheme.get_gap_score()};
+        int32_t gap_open{m_gap_scheme.get_gap_open_score() + gap_extension};
+
+        std::vector<int32_t> optimal_column{};
+        std::vector<int32_t> horizontal_column{};
+
+        // Initialise matrix.
+        optimal_column.clear();
+        horizontal_column.clear();
+        optimal_column.resize(sequence2.size() + 1, 0);
+        horizontal_column.resize(sequence2.size() + 1, 0);
+        int32_t diagonal{};
+        int32_t vertical{gap_open};
+
+        // Initialise the first column.
+        for (auto && [opt, hor] : views::zip(optimal_column, horizontal_column) | views::drop(1))
```

Comment:
I do not understand what the drop does here, are we then ignoring the first elements in optimal_column and horizontal_column, and if so why?

Reply:
Yes, I initiaist the first column. The first cell always is initialised with 0. So I just start with the next cell in the column, which is the first one that is initialised with gap costs.

Reply:
I added a comment.